### DISCUSSION TITLE
[DUVIDA] Coverage report no Vue.js

### DISCUSSION THREAD

**Discussion by WandersonAlves** [frontendbr/forum#599] [Perguntas e Respostas]
Olá pessoal!

Tava fazendo um mini projetinho em VueJS pra uma entrevista de emprego e após o termino da seleção, resolvi brincar um pouco mais com o projeto.

No momento estou fazendo os testes unitários de uma aplicação simples em VueJS porém não consigo gerar o coverage report para a mesma.

Alguém pode me dar uma luz?

A branch é essa: https://github.com/WandersonAlves/vue-crud-test/tree/unit-tests

Como eu vim do Angular, usei minha experiência de testes lá e tentei fazer tudo sozinho. Pesquisei um pouco aqui e ali e gerei esse karma.conf.js que na minha leitura, teoricamente estaria tudo ok, porém o report gerado não contempla os arquivos .vue muito menos os .js (os arquivos de store e rotas por exemplo). Os testes funcionam magicalmente bem.

Valeu galera!

**Comment by caioincau** [comment]
Se você usar o template webpack, do vue-cli, ele vem com lcov configurado.

Você pode usar ele de exemplo para configurar o lcov você mesmo:
https://github.com/vuejs-templates/webpack

**Comment by WandersonAlves** [comment]
Puts @icaioincau usei o webpack-simple :/ vou dar uma olhadinha, sou novo no webpack e ainda não saquei todo o poder dele (fazia tudo "manual" pelo gulp antigamente), valeu pela dica!

**Comment by infixremix** [comment]
@icaioincau What wifis down?

**Comment by WandersonAlves** [comment]
Consegui fazer a configuração, valeu @icaioincau na próxima uso o webpack em vez do webpack-simple. 

Bom que disso aprendi um pouco mais sobre o webpack

**Comment by caioincau** [comment]
Boa @WandersonAlves , webpack é tenso para aprender, não é nada humano como o gulp, mas depois que aprender, é só sair voando.
